src/main/java/minijava/ir/optimize/BaseOptimizer.java

```diff
@@ -4,33 +4,42 @@
 import firm.Graph;
 import firm.nodes.Node;
 import firm.nodes.NodeVisitor;
+import java.util.Collection;
+import minijava.ir.utils.FirmUtils;
 
 public abstract class BaseOptimizer extends NodeVisitor.Default implements Optimizer {
 
   protected Graph graph;
   protected boolean hasChanged;
 
   /**
-   * Implements the work-list algorithm.
+   * Implements the work-list algorithm, populating the initial worklist by passing .
    *
    * <p>Nodes in the work-list are processed by calling {@link Node#accept(NodeVisitor)
    * node.accept(this)}. In order for the data-flow analyses to be successful, implementations of
    * {@link BaseOptimizer} <b>must</b> set the protected {@code hasChanged} field to true in the
    * overwritten {@code visit(Node)} methods, if the outputs of the node's transfer function differ
    * from the outputs of the previous visiting.
    */
-  protected void fixedPointIteration() {
```

Comment:
Although the commit log might look scary, because there are so many of them, most are actually already on master and are an artifact of the squash and merge workflow. 

This is the most sigificant change for this PR, most of the the other file changes are just changing from chaotic iteration to use this new worklist approach where needed.